### PR TITLE
feat: implement project-wide proselintrc

### PR DESCRIPTION
**Pull request by carlsmedstad**
Hey again :wave: 

This PR makes proselint pick up `.proselintrc` file in the current working directory, i.e. project specific config files. Also included some refactoring and error reporting.

Resolves https://github.com/amperser/proselint/issues/1112.

### CODE REVIEW

**Comment by Nytelife26** [comment]
Thank you for the pull request! I'll triage & review now.

**Comment by Nytelife26** [comment]
@carlsmedstad can you fix the coverage issues by adjusting the relevant unit tests for your changes?

**Comment by carlsmedstad** [comment]
> @carlsmedstad can you fix the coverage issues by adjusting the relevant unit tests for your changes?

I extended one test., hpefully this increases the coverage enough. I'm not sure how to run the codecov job locally.

**Comment by suchow** [comment]
@Nytelife26 We’re getting build failures here and in some other PRs for reasons unrelated to the PR itself — Codecov is failing to upload results.

**Comment by Nytelife26** [comment]
> Codecov is failing to upload results.

I noticed this in another pull request. I am not entirely sure if Codecov was down briefly or not, but I'll try rerunning now a few hours later to see if it has been resolved yet.

**Comment by Nytelife26** [comment]
> hopefully this increases the coverage enough. I'm not sure how to run the codecov job locally.

You cannot, unfortunately. The best way to approach this is to check within Codecov's UI for the uncovered lines and add testing for them.

File: tests/test_config_flag.py [hunk 1]
```diff
@@ -10,20 +17,19 @@ def test_load_options_function():
...
     output = subprocess.run(["python", "-m", "proselint", "--demo", "--config",
                              "tests/test_config_flag_proselintrc"],
                             stdout=subprocess.PIPE, encoding='utf-8')
     assert "uncomparables.misc" not in output.stdout
+
+    output = subprocess.run(["python", "-m", "proselint", "--demo", "--config",
+                             "non_existing_file"],
+                            stderr=subprocess.PIPE, encoding='utf-8')
+    assert output.returncode == 2
+    assert 'FileNotFoundError' in output.stderr
```
Comment (carlsmedstad) [inline tests/test_config_flag.py:35]: > It appears that for this PR most of the coverage loss stems from proselint/command_line.py.
> To resolve the coverage issues, you need to:
> * Add testing that triggers the sys.exit(2) statement
> * Add testing for if the configuration file is not a file
>
> I will adjust the Codecov settings to set a threshold for coverage loss until then.

Comment (Nytelife26) [reply tests/test_config_flag.py:35]: Apologies, I didn't see this until just now. I'll investigate later tonight.

Comment (carlsmedstad) [reply tests/test_config_flag.py:35]: No worries! Feel free to ping me if you need me for anything.

**Comment by Nytelife26** [comment]
@carlsmedstad As it happens, you were right - there was an issue with the way we were testing it. Python coverage does not pick up `subprocess` calls, so switching to Click's unit testing fixed it. Thanks for the PR - we're now ready for review.

**Review by Nytelife26 — APPROVED** [review]

**Review by suchow — APPROVED** [review]

**Pull request merged**